ykrt/src/compile/jitc_yk/codegen/x64/mod.rs

```diff
@@ -1979,6 +1979,10 @@ impl<'a> Assemble<'a> {
         let dest_type = self.m.type_(i.dest_tyidx());
         let dest_bitsize = dest_type.bit_size().unwrap();
 
+        if src_bitsize == 32 && dest_bitsize == 64 {
```

Comment:
Why is this safe?

Don't we need to zero the upper 32-bits?

Reply:
Hmm, yes, if we had a left shift before, I suppose we need to.

I suggest I remove this commit during a rebase.

Reply:
It's more about what follows. If the `zext` instruction is followed by for example, and `add`, then the undefined bits could cause an incorrect result (if any bits are set up there).

> I suggest I remove this commit during a rebase.

OK

Reply:
> It's more about what follows.

Yes, we agree.